### ISSUE TITLE
Error checkDebugDuplicateClasses (Amplify Push Notifications with react native 0.61.2 for Android)

### ISSUE THREAD

**Issue by platon1993** [aws-amplify/amplify-js#4593]
**Describe the bug**
I try to install Amplify Push Notifications with react native 0.61.2 for Android. There is no any information for this in documentation for this version of React native.
I have an error:  
Task :app:checkDebugDuplicateClasses FAILED
Execution failed for task ':app:checkDebugDuplicateClasses'.
> 1 exception was raised by workers:
  java.lang.RuntimeException: Duplicate class com.google.android.gms.gcm.PendingCallback found in modules classes.jar (com.firebase:firebase-jobdispatcher:0.6.0) and classes.jar (com.google.android.gms:play-services-gcm:16.1.0)

**To Reproduce**

1. Install new React Native
2. npm install @aws-amplify/pushnotification
3. Set up settings on my build.gradle, app/build.gradle, AndroidManifest.xml

android/build.gradle

> // Top-level build file where you can add configuration options common to all sub-projects/modules.
> 
> buildscript {
>     ext {
>          buildToolsVersion = "28.0.3"
>                 minSdkVersion = 16
>                 compileSdkVersion = 28
>                 targetSdkVersion = 28
>                 supportLibVersion = "28.0.0"
>                 googlePlayServicesVersion = "16.1.0" // default: "+"
>                 firebaseVersion = "17.3.4" // default: "+"
>     }
>     repositories {
>         google()
>         jcenter()
>     }
>     dependencies {
>         classpath("com.android.tools.build:gradle:3.4.2")
>         classpath 'com.google.gms:google-services:4.3.2'
> 
>         // NOTE: Do not place your application dependencies here; they belong
>         // in the individual module build.gradle files
>     }
> }
> 
> allprojects {
>     repositories {
>         mavenLocal()
>         maven {
>             // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
>             url("$rootDir/../node_modules/react-native/android")
>         }
>         maven {
>             // Android JSC is installed from npm
>             url("$rootDir/../node_modules/jsc-android/dist")
>         }
> 
>         google()
>         jcenter()
>         maven { url 'https://jitpack.io' }
>     }
> }

> app/build.gradle
> 
> 
> apply plugin: "com.android.application"
> 
> import com.android.build.OutputFile
> 
> 
> project.ext.react = [
>     entryFile: "index.js",
>     enableHermes: false,  // clean and rebuild if changing
> ]
> 
> apply from: "../../node_modules/react-native/react.gradle"
> 
> def enableSeparateBuildPerCPUArchitecture = false
> 
> 
> def enableProguardInReleaseBuilds = false
> 
> def jscFlavor = 'org.webkit:android-jsc:+'
> 
> def enableHermes = project.ext.react.get("enableHermes", false);
> 
> android {
>     compileSdkVersion rootProject.ext.compileSdkVersion
> 
>     compileOptions {
>         sourceCompatibility JavaVersion.VERSION_1_8
>         targetCompatibility JavaVersion.VERSION_1_8
>     }
> 
>     defaultConfig {
>         applicationId "com.cyklonbonus"
>         minSdkVersion rootProject.ext.minSdkVersion
>         targetSdkVersion rootProject.ext.targetSdkVersion
>         versionCode 1
>         versionName "1.0"
>     }
>     splits {
>         abi {
>             reset()
>             enable enableSeparateBuildPerCPUArchitecture
>             universalApk false  // If true, also generate a universal APK
>             include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
>         }
>     }
>     signingConfigs {
>         debug {
>             storeFile file('debug.keystore')
>             storePassword 'android'
>             keyAlias 'androiddebugkey'
>             keyPassword 'android'
>         }
>     }
>     buildTypes {
>         debug {
>             signingConfig signingConfigs.debug
>         }
>         release {
>             // Caution! In production, you need to generate your own keystore file.
>             // see https://facebook.github.io/react-native/docs/signed-apk-android.
>             signingConfig signingConfigs.debug
>             minifyEnabled enableProguardInReleaseBuilds
>             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
>         }
>     }
>     // applicationVariants are e.g. debug, release
>     applicationVariants.all { variant ->
>         variant.outputs.each { output ->
>             // For each separate APK per architecture, set a unique version code as described here:
>             // https://developer.android.com/studio/build/configure-apk-splits.html
>             def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
>             def abi = output.getFilter(OutputFile.ABI)
>             if (abi != null) {  // null for the universal-debug, universal-release variants
>                 output.versionCodeOverride =
>                         versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
>             }
> 
>         }
>     }
> }
> 
> dependencies {
>   implementation project(':@aws-amplify_pushnotification')
>     implementation fileTree(dir: "libs", include: ["*.jar"])
>     implementation "com.facebook.react:react-native:+"  // From node_modules
> 
> 
>     if (enableHermes) {
>         def hermesPath = "../../node_modules/hermes-engine/android/";
>         debugImplementation files(hermesPath + "hermes-debug.aar")
>         releaseImplementation files(hermesPath + "hermes-release.aar")
>     } else {
>         implementation jscFlavor
>     }
> }
> 
> // Run this once to be able to run the application with BUCK
> // puts all compile dependencies into folder libs for BUCK to use
> task copyDownloadableDepsToLibs(type: Copy) {
>     from configurations.compile
>     into 'libs'
> }
> 
> apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
> apply plugin: 'com.google.gms.google-services'

AndroidManifest.xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.cyklonbonus">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <permission
            android:name="${applicationId}.permission.C2D_MESSAGE"
            android:protectionLevel="signature" />
    <uses-permission android:name="${applicationId}.permission.C2D_MESSAGE" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

    <application
      android:name="android.support.multidex.MultiDexApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">

        
        
        <service
                android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationMessagingService">
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT"/>
            </intent-filter>
        </service>
        <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
        
        
        <service
                android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationDeviceIDService">
            <intent-filter>
                <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
            </intent-filter>
        </service>
        <receiver
                android:name="com.amazonaws.amplify.pushnotification.modules.RNPushNotificationBroadcastReceiver"
                android:exported="false" >
            <intent-filter>
                <action android:name="com.amazonaws.amplify.pushnotification.NOTIFICATION_OPENED"/>
            </intent-filter>
        </receiver>
        

        <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:screenOrientation="portrait"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

    </application>

</manifest>

**Expected behavior**
Success build

**Comment by Ashish-Nanda** [comment]
Hi @platon1993 

Please take a look at this [comment]([url](https://github.com/aws-amplify/amplify-js/issues/4728#issuecomment-575563945)) on a related issue. It contains a link to a recent article detailing the steps for setting up push notifications on android. The React Native version being used there is 0.61.5 based on the linked github [repo](https://github.com/dantasfiles/AmplifyAndroidPush).

We will work on updating our docs soon for newer versions of React Native.

**Comment by bneigher** [comment]
I am having this issue since upgrading @aws-amplify/pushnotification to 2.x.x

In particular, before it was autolinked, the only way to get around the duplicates symbols error was by having this:

```
dependencies {
   implementation (project(':@aws-amplify_pushnotification')){
      exclude group: 'com.firebase'
      exclude group: 'com.google.android.gms', module: 'play-services-gcm'
   }
}
```

After the upgrade, I can no longer reference the project here, thus I can no longer specify the exclude group.

**Comment by mauerbac** [comment]
hey @bneigher - sorry for the delay here. Would you mind re-testing with the latest push notification version ? I'd like to help you but need more information to debug or let us know if this is no longer an issue. 

Thank you

**Comment by bneigher** [comment]
@mahayash315 don't worry about it.. I've had too many issues with amplify on react-native and I've since moved everything out of my app.  Pinpoint has really been a painpoint 😅

**Comment by mauerbac** [comment]
Sorry to hear that. Any feedback to pass along? Will close for now.

**Comment by the-smart-home-maker** [comment]
Hi there, 

I am facing exactly the same problem reported here since lately upgrading my packages to the latest version (`yarn upgrade`). 

I can't get my React Native app to run on Android any more as of the following error:
```
java.lang.RuntimeException: Duplicate class com.google.android.gms.gcm.PendingCallback found in modules classes.jar (com.firebase:firebase-jobdispatcher:0.6.0) and classes.jar (com.google.android.gms:play-services-gcm:16.1.0)
```
Is there any solution on the way? Unfortunately this issue here has been closed even though there seems to be no solution yet.

I have also raised a question on StackOverflow (https://stackoverflow.com/questions/67782807/react-native-android-duplicate-class-com-google-android-gms-gcm-pendingcallback) about this issue as I was not sure if I had a wrong configuration or if the problem is caused by the aws amplify library but now I am pretty convinced that it is a bug.

Is there any solution to be expected soon?
